test/L2/L2HodlerdropRedistribution.t.sol

```diff
@@ -0,0 +1,566 @@
+    }
+
+    function test_SendLSKToEcosystemFundWallet_AirdropV2PeriodNotOver() public {
+        // proceed time to HODLERDROP_REDISTRIBUTION_DURATION so that hodlerdrop-redistribution period is not over
+        vm.warp(block.timestamp + l2HodlerdropRedistribution.HODLERDROP_REDISTRIBUTION_DURATION() * 1 days);
+
+        vm.expectRevert("L2HodlerdropRedistribution: hodlerdrop-redistribution is not over yet");
+        l2HodlerdropRedistribution.sendLSKToEcosystemWallet();
+    }
+
+    function test_SendLSKToEcosystemFundWallet_OnlyOwner() public {
+        // alice tries to send all L2LiskToken to Ecosystem Fund wallet
+        vm.prank(alice);
+        vm.expectRevert(abi.encodeWithSelector(Ownable.OwnableUnauthorizedAccount.selector, alice));
+        l2HodlerdropRedistribution.sendLSKToEcosystemWallet();
+    }
+
+    function aliceSatifiesStakingTier1() internal {
+        // alice stakes 30 and 50 L2LiskToken for minimum and minumum plus 1 days respectively in two positions
+        vm.startPrank(alice);
+        l2Staking.lockAmount(alice, 30 * 10 ** 18, l2HodlerdropRedistribution.MIN_STAKING_DURATION_TIER_1());
+        l2Staking.lockAmount(alice, 50 * 10 ** 18, l2HodlerdropRedistribution.MIN_STAKING_DURATION_TIER_1() + 1);
+        vm.stopPrank();
+
+        // maximum staking tier 1 hodlerdrop-redistribution amount for alice is 80 L2LiskToken
+        assertEq(l2HodlerdropRedistribution.satisfiesStakingTier1(alice, 80 * 10 ** 18), true);
+
+        // check that bigger amount than 80 L2LiskToken does not satisfy staking tier 1
+        assertEq(l2HodlerdropRedistribution.satisfiesStakingTier1(alice, (80 * 10 ** 18) + 1), false);
+    }
+
+    function test_SatisfiesStakingTier1() public {
+        // check that alice satisfies staking tier 1
+        aliceSatifiesStakingTier1();
+    }
+
+    function test_charlieSatifiesStakingTier1With1000LockingPositionsAndClaim() public {
+        vm.startPrank(charlie);
+        // charlie stakes 1 L2LiskToken for minimum days in 400 positions
+        for (uint256 index = 0; index < 1000; index++) {
+            l2Staking.lockAmount(charlie, 1 * 10 ** 18, l2HodlerdropRedistribution.MIN_STAKING_DURATION_TIER_1());
+        }
+        vm.stopPrank();
+
+        // maximum staking tier 1 hodlerdrop-redistribution amount for charlie is 400 L2LiskToken
+        assertEq(l2HodlerdropRedistribution.satisfiesStakingTier1(charlie, 1000 * 10 ** 18), true);
+
+        uint256 charlieBalanceBefore = l2LiskToken.balanceOf(charlie);
+        bytes32[] memory merkleProof = new bytes32[](2);
+        merkleProof[0] = bytes32(0x2d5b63a817daa1152c66e8b254400766fac6b7958a2f8afff1dea8e35e84f013);
+        merkleProof[1] = bytes32(0xf0df3dcda05b4fbd9c655cde3d5ceb211e019e72ec816e127a59e7195f2cd7f5);
+        l2HodlerdropRedistribution.claimHodlerdrop(charlie, 1000 * 10 ** 18, merkleProof);
+        assertEq(l2LiskToken.balanceOf(charlie), charlieBalanceBefore + 500 * 10 ** 18);
+        assertGe(gasleft(), 100);
```

Comment:
Could we also add `gasleft()` at the beginning of the unit test and calculate the gas difference at the end?
The difference should not exceed 30 million, as this is the current block gas limit on Lisk.

Reply:
If we are left with `100` in the current unit tests, does this mean that `1000` locking positions would be our maximum?

Reply:
refactor the test to check the gas for the claim func to be less than our gas limit per block.